src/NUnitFramework/framework/Internal/Builders/NUnitTestCaseBuilder.cs

```diff
@@ -136,7 +136,14 @@ private static bool CheckTestMethodSignature(TestMethod testMethod, ParameterSet
 
             parameters = testMethod.Method.GetParameters();
 
-            int argsNeeded = parameters.Length;
+            int minArgsNeeded = 0;
+            foreach (var parameter in parameters)
+            {
+                // IsOptional is supported since .NET 1.1 
+                if (!parameter.IsOptional)
+                    minArgsNeeded++;
+            }
```

Comment:
ParameterInfo.IsOptional is not available under CF 3.5. This whole sequence should be conditional, with minArgsNeeded set to the same as maxArgsNeeded under NETCF.